netket/driver/steady_state.py

```diff
@@ -121,7 +121,15 @@ def _forward_and_backward(self):
 
         # if it's the identity it does
         # self._dp = self._loss_grad
-        self._dp = self.preconditioner(self.state, self._loss_grad)
+        try:
+            self._dp = self.preconditioner(
+                self.state, self._loss_grad, step_value=self.step_count
+            )
+        except TypeError:
+            warn_deprecation(
+                "Preconditioners should accept an optional `step_value` argument."
+            )
+            self._dp = self.preconditioner(self.state, self._loss_grad)
```

Comment:
Can you make this a function defined in `netket/driver/vmc_common.py` and use it in vmc/steadystate/ecc so it's only defined once instead of several times?

And then, add a test checking both branches of this function.